### PR TITLE
Add verifiableURI contentType in LSP2

### PR DESCRIPTION
**Pull request by YamenMerhi**

### CODE REVIEW

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 1]
```diff
@@ -633,7 +635,88 @@ Setting multiple permissions like `TRANSFER VALUE + CALL + SET DATA` will result
...
+verification: {
+   method:
+   data:
+   source:
+}
+```
+
+The verification object consisting of three fields:
+
+- **method**: This field specifies the method of verification. Verification can vary being hash reference, signature reference, etc .. Some methods that can be used for example include `'keccak256(utf8)'`, where the verification relies on hashing the UTF-8 encoded content, or `'ecdsa'`, indicating that the verification uses the ECDSA signature scheme.
```
Comment (JeneaVranceanu) [inline LSPs/LSP-2-ERC725YJSONSchema.md:654]:
```suggestion
- **method**: This field specifies the method of verification. Verification can vary between hash reference, signature reference, etc. Some methods that can be used, for example, include `'keccak256(utf8)'`, where the verification relies on hashing the UTF-8 encoded content, or `'ecdsa'`, indicating that the verification uses the ECDSA signature scheme.
```

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 2]
```diff
@@ -633,7 +635,88 @@ Setting multiple permissions like `TRANSFER VALUE + CALL + SET DATA` will result
...
+
+**VerifiedURL** is a concept aimed at ensuring the authenticity and integrity of content stored. This system contains two primary components:
+
+#### 1. **The Verification Object**
+
+```js
+verification: {
+   method:
+   data:
+   source:
```
[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply LSPs/LSP-2-ERC725YJSONSchema.md:648]: Maybe the `data` could be "generic" depending on the `method`.

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 3]
```diff
@@ -633,7 +635,88 @@ Setting multiple permissions like `TRANSFER VALUE + CALL + SET DATA` will result
...
+
+The URL is the link to the content being verified.
+
+<hr>
+
+**Example: Constructing a VerifiedURL**
+
+The creation of a VerifiedURL involves the concatenation of various elements:
+
+1. **verification method**: First 4 bytes of the `keccak256('method')`.
```
Comment (JeneaVranceanu) [inline LSPs/LSP-2-ERC725YJSONSchema.md:670]:
```suggestion
1. **verification method**: First 4 bytes of the hash of the selected method (e.g. `keccak256('method')`).
```

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 4]
```diff
@@ -633,7 +635,88 @@ Setting multiple permissions like `TRANSFER VALUE + CALL + SET DATA` will result
...
+2. **verification data and source length indicator**: 2 bytes indicating the length of the data and the source fields.
```
Comment (JeneaVranceanu) [inline LSPs/LSP-2-ERC725YJSONSchema.md:671]: I'm not sure I understood it correctly but I'll suggest a slight modification that will probably make it clearer:
```suggestion
2. **verification data and source length indicator**: 2 bytes indicating the combined bytes count of the data and the source fields.
```

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 5]
```diff
@@ -633,7 +635,88 @@ Setting multiple permissions like `TRANSFER VALUE + CALL + SET DATA` will result
...
+> '0x696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178'
+
+
+// final result (to be stored on chain)
+const VerifiedURL = verificationMethod  +  verificationDatalength.substring(2) +  verificationData.substring(2) + url.substring(2)
+                    ^                      ^                                      ^                   ^
+                    0x6f357c6a          +  0020                                +  820464ddfac1be... + 696670733a2f2...
+
+// structure of the VerifiedURL
+0x6f357c6a + 0020 +  820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361 + 696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178
```
Comment (JeneaVranceanu) [inline LSPs/LSP-2-ERC725YJSONSchema.md:711]: 🤷🏻‍♂️ 😅 
```suggestion
0x6f357c6a + 0020 + 820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361 + 696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178
```

**Review by JeneaVranceanu — COMMENTED** [review]
LGTM, just a few comments.

File: LSPs/LSP-2-ERC725YJSONSchema.md [hunk 6]
```diff
@@ -32,6 +32,7 @@ requires: ERC725Y
   - [Tuples of `valueType`](#tuples-of-valuetype)
 - [ValueContent](#valuecontent)
   - [BitArray](#bitarray)
+  - [VERIFIEDURL](#verifiedurl)
```
[the remark this answers is not in the document]
Comment (frozeman) [reply LSPs/LSP-2-ERC725YJSONSchema.md:35]: no becuase that's the link

[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply LSPs/LSP-2-ERC725YJSONSchema.md:35]: So then it should be the following? ->

```suggestion
  - [VerifiedURL](#verifiedurl)
```

**Review by JeneaVranceanu — APPROVED** [review]

**Pull request merged**